Add FluentD DaemonSet Chart

rendhalver opened the pull request:
Sets up a FluentD DaemonSet to gather and send container logs to a logging service
Elasticsearch is setup as the default but that is easily changable

This is a rewrite of #3232 to make it more configurable for multiple logging sources.

*****************************************************************************************

Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the 
history. This will make it easier to identify new changes. The PR will be squashed 
anyways when it is merged. Thanks.

*****************************************************************************************

Commit:
Add FluentD DaemonSet Chart

Sets up a FluentD DaemonSet to gather and send container logs to a logging service
Elasticsearch is setup as the default but that is easily changable

kenden commented:
Thanks for the PR @rendhalver.
Having one fluentd chart for elasticsearch/cloudwatch/... is awesome.
I have no access to a computer at the moment, but I think it would be nice to know how to switch elasticsearch to, say, cloudwatch in the `README.md`.

Commit:
Add annotations to restart daemonset when configmap changes

Add config to add pullSecrets as needed
Fix configmap to include the right files

rendhalver commented:
Closing in favour of #3379
